### ISSUE TITLE
Renewal of funds

### ISSUE THREAD

**Issue by maxvisser** [teamforus/general#252]
Learn more about improvement proposals: https://bit.ly/2xLJT3R

## Impacted areas:

- [x] Design
- [x] Code
- [ ] Infrastructure
- [x] Users
- [ ] Finances
- [ ] Processes

## Current situation
Fund are recreated and configured

## Desired situation
Fund is renewed.

Sponsor presses renew after fund is closed. Old fund will connect as a child to the new fund. 

Nice to have: criteria of fund is copypasted, providers are invited automatically

## Plan
-

**Comment by jamalv** [comment]
This is a very great CR: could you please index the scope / impact? If it is literally a button and the user can do it themselves, this would save a lot of work for all the funds that need to be renewed next year.

On a related note: I remember a discussion we had about how we prevent "ghost-town" webshops. One of the strategies we had, is to let providers re-apply; a better solution we came up with was a "last-seen" I opened a separate CR for that: https://github.com/teamforus/general/issues/282

**Comment by maxvisser** [comment]
Currently the record_types need to be manually created as most of the fund use an Eligible key which is unique for that year. If we add time sensitivity to records we don't need to renew meedoen_2020_eligle to meedoen_2021_eligible each year. I would propose to add time sensitivity first to the criteria system before starting on renewal.

**Comment by maxvisser** [comment]
There is aswell a use case to not ask the requester to apply again and renew their voucher based on new prevalidation or voucher generator

**Comment by maxvisser** [comment]
Scope too big for estimation. I would split this CR up into atleast three seperate requests:
- renewal for provider -> add last seen provider automatically to your fund
- renewal for sponsor -> criteria is added again and statistics of old fund can be compared with new fund
- renewal for requester -> don't need to apply again if I am still eligible.

**Comment by jamalv** [comment]
unnasigning as this is on hold

**Comment by maxvisser** [comment]
Idea for this CR:
- Let sponsor extend end date 1 week before end date is reached.
- A requester can get a second voucher if previous voucher is expired (would be nice to make this configurable)

**Comment by jamalv** [comment]
Okay, let's see what is still realistic to do do for this year. What I'm most interested in are scope:medium CR's that are still be feasable to be released within 2 sprints and have a positive impact on our users/customers.

@maxvisser any ideas? :)

**Comment by maxvisser** [comment]
So how this would work is:
Currently how our system works a fund run have a start and end date and vouchers have an expire date.

Some problems with the current system:
Requester who applied late (december) needs to reapply again in a few months
Funds need to be reconfigured by Forus
Providers need to reapply

For each problem there are different fixes. Lets go through them all

**Requester who applied late needs to reapply again in a few months**
Allow sponsor to set a time of validity on a criteria (for example kindpakket_eligible = ja should be validated in within the past year or user needs to be validated again) This results in that the criteria could reused.

**Funds need to be reconfigured by Forus**
Extend the expire date of the fund by the sponsor, allowing for example Nijmegen to extend the fund by one year in december; Or automatically extend the fund by 1 month (each month). This results in voucher created in that month being valid one extra month. For example voucher generated in december are valid december april next year. 

**Providers need to reapply**
A solution is to make it easier for the provider to accept a reservation and make it easier for a sponsor to invite the provider; Resulting in providers having less hassle to continue using the system

I think only the last point needs design others are backend fixes. Ofcourse we need to check if these are the only fixes.
